src/ai/defaultai.cc

```diff
@@ -5646,10 +5602,28 @@ BuildingNecessity DefaultAI::check_building_necessity(BuildingObserver& bo,
 				}
 			}
 			if (site_needed_for_economy == BasicEconomyBuildingStatus::kEncouraged) {
-				tmp_score += 4;
+				tmp_score += 10 + std::abs(management_data.get_military_number_at(160));
 			}
 			if (site_needed_for_economy == BasicEconomyBuildingStatus::kDiscouraged) {
-				tmp_score -= 2;
+				tmp_score -= std::abs(management_data.get_military_number_at(178));
+			}
+
+			// now we find out if the building is needed depending on output stocklevel
+			const uint32_t current_stocklevel = (get_stocklevel(bo, gametime));
+			const uint8_t stocklevel_threshhold =
+			   std::abs(management_data.get_military_number_at(180));
+
```

Comment:
Is not a range 0-100 too broad? Maybe reduce it to 5-55 or so....

Reply:
well, I have no problem with the upper limit, cause this should be clearly dependent on training however a lower limit of 0 might be too low, so I think I'll simply add 20 to the threshhold resulting in 20-120

Reply:
I would go for lower, but can be.

Reply:
ok i'll change to 10

Reply:
thanks